src/lib/sqlite/sqliterg.test.ts

```diff
@@ -161,6 +170,57 @@ describe('remote database', () => {
       })
     }
   })
+
+  it('select columns order', async () => {
+    // IMPORTANT: this test uses a "patched" version of `ws4sqlite` that
+    // ensures returned object keys' order (`thanhnguyen2187/ws4sqlite`).
+    //
+    // Look at https://github.com/proofrock/ws4sqlite/pull/38 for more
+    // information.
```

Comment:
Problem with `w4sqlite`: the library does not allow unnamed parameters (`INSERT INTO ... VALUES (?, ?, ?)`), which breaks almost every other queries. Related issue: https://github.com/proofrock/ws4sqlite/issues/40.

Reply:
The problem should be resolved after https://github.com/proofrock/ws4sqlite/pull/41 getting merged. `dataManager` is able to "combine" local snippets and remote snippets successfully. When we modify a synchronized snippet "locally", the natural thing is to modify the remote snippet as well. There might be a case where there is the third device that writes at the same time: since the "local remote" data is not updated, the conflict status only shows upon restart.

Reply:
A few things to think about for the way `dataManager` is periodically checking and generating synchronization state:

- On creating a new snippet locally: it works fine.
- On updating a synchronized snippet: it would not work since the data is updated locally and not remotely. It would create a "conflicted" state.
- On deleting a synchronized snippet: it would not work either. The snippet's state would become "remote-only", and the data gets recreated.

Resolving the issue is quite simple: whenever we do upserting locally, also do the same remotely. Concurrent writes should not be a problem, as the data is stored locally and then conflicts can be resolved manually anyway.

The problem is with the current code structure: `card-v2.svelte` uses `localSnippetsStore` for data creating and updating. Modifying it to `dataManager` can be quite the big change.